store/src/java/com/zimbra/cs/service/mail/RecoverAccount.java

```diff
@@ -90,20 +90,22 @@ public Element handle(Element request, Map<String, Object> context) throws Servi
 
         RecoverAccountResponse resp = new RecoverAccountResponse();
         switch (op) {
-            case GET_RECOVERY_EMAIL:
-                recoveryEmail = StringUtil.maskEmail(recoveryEmail);
-                ZimbraLog.account.debug("%s Recovery email: %s", LOG_OPERATION, recoveryEmail);
-                resp.setRecoveryEmail(recoveryEmail);
-                break;
-            case SEND_RECOVERY_CODE:
-                String storedCodeString = user.getResetPasswordRecoveryCode();
-                ZonedDateTime currentDate = ZonedDateTime.now(ZoneId.systemDefault());
-                Map<String, String> recoveryCodeMap = new HashMap<String, String>();
+        case GET_RECOVERY_EMAIL:
+            recoveryEmail = StringUtil.maskEmail(recoveryEmail);
+            ZimbraLog.account.debug("%s Recovery email: %s", LOG_OPERATION, recoveryEmail);
+            resp.setRecoveryEmail(recoveryEmail);
+            break;
+        case SEND_RECOVERY_CODE:
+            String storedCodeString = user.getResetPasswordRecoveryCode();
+            ZonedDateTime currentDate = ZonedDateTime.now(ZoneId.systemDefault());
+            Map<String, String> recoveryCodeMap = new HashMap<String, String>();
+            int maxAttempts = user.getPasswordRecoveryMaxAttempts();
```

Comment:
How would this get reset?

Reply:
Once max attempts is reached, feature is in suspended state.It can be reset by admin or when request is triggered after suspension time is over.